packages/EasyBatch/src/Transformers/AbstractBatchObjectTransformer.php

```diff
@@ -86,6 +87,12 @@ protected function doTransformToArray(BatchObjectInterface $batchObject): array
 
     protected function serialize(object $message): string
     {
+        if ($message instanceof HandlerFailedException) {
```

Comment:
@alexndlm nice job identifying this issue!
However this class isn't part of the Symfony Bridge for this package, so it doesn't make sense to have specific logic for it here. When implementing this serialize/unserialize logic, I thought it was simple enough to keep it in the transformer directly, but based on this issue, it might be time to extract it to its own service so we could safely fix this bug and not couple the core logic of the package with a specific framework.

Here is the way I see it working:

- Create a serializer interface for this package which defines both `serialize` and `unserialize` methods
- Create its default implementation and copy/paste the current logic from the transformer
- Inject this serializer interface into the transformer, and use it instead of the current private methods
- Create a Symfony specific implementation of the serializer interface which decorates the default one and implements your logic
- Register the default implementation as a service using a specific ID (defined as part of the bridge interface)
- Register the Symfony specific implementation to decorate the default one

What do you think?

Reply:
Thanks for the advice!
Done.